src/main/java/com/bookshop/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,40 @@
+package com.bookshop.repository.impl;
+
+import com.bookshop.exception.BookNotFoundException;
+import com.bookshop.exception.BookNotSavedException;
+import com.bookshop.model.Book;
+import com.bookshop.repository.BookRepository;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired

```

Comment:
As of Spring Framework 4.3, an @Autowired annotation on such a constructor is no longer necessary if the target bean defines only one constructor to begin with. See [note here ](https://docs.spring.io/spring-framework/reference/core/beans/annotation-config/autowired.html)Let’s check it ;)